dali/python/nvidia/dali/pipeline.py

```diff
@@ -1324,7 +1324,9 @@ def create_pipeline(*args, **kwargs):
                 pipe.set_outputs(*po)
             return pipe
         return create_pipeline
-    return actual_decorator(fn) if fn else actual_decorator
+    ret = actual_decorator(fn) if fn else actual_decorator
+    ret.is_pipeline_def = True
```

Comment:
I didn't do very extensive research in this topic, however this pattern seems to be popular enough. TBH, I can't imagine a scenario, where `is_pipeline_def` could be abused.

And if it should be private? I guess the most common way it could be used is outside the `pipeline_def` body, which rather gives the incentive to make it public. After all, why a regular user shouldn't be allowed to use it?